packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,371 @@
+        const x = node.offset.x;
+        const y = node.offset.y;
+        // Flip Z axis to convert handedness.
+        const z = -node.offset.z;
+        return Matrix.Translation(x, y, z);
+    }
+
+    /**
+     * Creates animations for the BVH node
+     * @param node - The BVH node to create animations for
+     * @param context - The loader context
+     * @returns The created animations
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    private static _createAnimations(node: BVHNode, context: LoaderContext): Animation | null {
+        if (node.frames.length === 0) {
+            return null;
+        }
+
+        const keyFrames: IAnimationKey[] = [];
+
+        // Create position animation if there are position channels
+        const hasPosition = node.channels.some((c) => c === BVHLoader.X_POSITION || c === BVHLoader.Y_POSITION || c === BVHLoader.Z_POSITION);
+
+        // Create rotation animation if there are rotation channels
+        const hasRotation = node.channels.some((c) => c === BVHLoader.X_ROTATION || c === BVHLoader.Y_ROTATION || c === BVHLoader.Z_ROTATION);
+
+        for (let i = 0; i < node.frames.length; i++) {
+            const frame = node.frames[i];
+            const time = i * context.frameRate * 1000; // Convert to milliseconds
+
+            if (hasPosition || hasRotation) {
+                let matrix = Matrix.Identity();
+
+                if (hasRotation) {
+                    const rotationMatrix = new Matrix();
+                    frame.rotation.toRotationMatrix(rotationMatrix);
+                    matrix = rotationMatrix;
+                }
+
+                if (hasPosition) {
+                    const position = frame.position;
+                    matrix.setTranslation(new Vector3(position.x, position.y, position.z));
+                }
+
+                keyFrames.push({
+                    frame: time,
+                    value: matrix,
+                });
+            }
+        }
+
+        if (keyFrames.length === 0) {
+            return null;
+        }
+
+        const fps = 60 / context.frameRate;
+        const animation = new Animation(node.name + "_anim", "_matrix", fps, Animation.ANIMATIONTYPE_MATRIX, Animation.ANIMATIONLOOPMODE_CYCLE);
+        animation.setKeys(keyFrames);
+
+        return animation;
+    }
+
+    /*
+         Recursively parses the HIERARCHY section of the BVH file
+            
+         - lines: all lines of the file. lines are consumed as we go along.
+         - firstLine: line containing the node type and name e.g. "JOINT hip"
+         - parent: the parent node for hierarchy
+         - context: the loader context containing the list of nodes and other data
+        
+         returns: a BVH node including children
+        */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    protected static _readNode(lines: string[], firstLine: string, parent: BVHNode, context: LoaderContext): BVHNode {
```

Comment:
Any reason two of these functions are protected?

Reply:
No, just changed to private! Thank you again!